ext/node/ops/util.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+use deno_core::error::AnyError;
+use deno_core::op2;
+use deno_core::OpState;
+use deno_core::ResourceHandle;
+use deno_core::ResourceHandleFd;
+
+#[repr(u32)]
+enum HandleType {
+  #[allow(dead_code)]
+  Tcp = 0,
+  Tty,
+  #[allow(dead_code)]
+  Udp,
+  File,
+  Pipe,
+  Unknown,
+}
+
+#[op2(fast)]
+pub fn op_node_guess_handle_type(
+  state: &mut OpState,
+  rid: u32,
+) -> Result<u32, AnyError> {
+  let handle = state.resource_table.get_handle(rid)?;
+  let handle_type = match handle {
+    ResourceHandle::Fd(handle) => guess_handle_type(handle),
+    _ => HandleType::Unknown,
+  };
```

Comment:
Iif you test `handle.is_terminal` here and early-return `Tty`, you can get rid of a lot of the unsafe code below.